tests/test_observatory_envar.py

```diff
@@ -0,0 +1,55 @@
+import os
+
+import pytest
+
+
+@pytest.fixture
+def sandbox(tmp_path):
+    class Sandbox:
+        pass
+
+    o = Sandbox()
+    e = os.environ.copy()
+    try:
+        del os.environ["PINT_OBS_OVERRIDE"]
+    except KeyError:
+        pass
+
+    o.override_dir = tmp_path / "override"
```

Comment:
You can probably just use o.override_dir=tmp_path, it's a fresh-generated empty directory.

Reply:
I'm unsure of the right way to do this.  If I import `pint.observatory.observatories` in the sandbox creation (to save the registry), then when I run import again in the test it will recognize it as already imported and won't re-read it, correct?  On the other hand if I just import in the test function then I don't have a way to save the registry.

I guess I could ask if this particular test is necessary.  there are other tests that demonstrate overriding existing observatories.  the unique feature here is that it happens through the environment variable.  I agree it would be good to check that but it's causing me some trouble.

Reply:
OK, I think I figured it out.  I now have a test for the correct and incorrect (overridden) GBT in `test_observatory_envar.py`, and the correct GBT in the existing `test_observatory.py`.  The registry is preserved and there's also a method now to clear it.

Reply:
This is why making stuff happen on import is a pain in the neck - the import actually happens once per test run, the first time the module is imported. It may make sense to have the module just call a function (no conditionals, nothing) and that function (which you can call from a test) do any checking of environment variables and multiple opening. Then testing that function is more feasible. It also opens a path to calling that function when the data is needed rather than on import, if we decide to do that.

Reply:
Yes, I agree.  I was considering doing just that but wanted to wait for your clock PR to get merged first.